File: app/src/main/java/com/rachit/tripathi75/soundspot/classes/DissolvePageTransformer.java
package com.rachit.tripathi75.soundspot.classes;

import android.view.View;

import androidx.viewpager2.widget.ViewPager2;

public class DissolvePageTransformer implements ViewPager2.PageTransformer {
    @Override
    public void transformPage(View page, float position) {
        if (position <= 0) {
            page.setAlpha(1 - position);
        } else {
            page.setAlpha(0f);
        }
    }
}

File: app/src/main/java/com/example/soundspot/classes/DissolvePageTransformer.kt
package com.example.soundspot.classes

import android.view.View
import androidx.viewpager2.widget.ViewPager2

class DissolvePageTransformer : ViewPager2.PageTransformer {
    override fun transformPage(page: View, position: Float) {
        if(position <= 0) {
            page.alpha = 1 - position
        }
        else {
            page.alpha = 0f
        }
    }
}